### DISCUSSION TITLE
Backpack Feature Plans for Q4 2023

### DISCUSSION THREAD

**Discussion by tabacitu** [Laravel-Backpack/community-forum#660] [Team Talk]
Back in Nov 2021, some our users have [voted on "what we should build next"](https://web.archive.org/web/20220205220227/https://github.com/Laravel-Backpack/CRUD/discussions/3960). Let's list the winners:

| Votes | Feature Request                                                       | Status |
|-------|----------------------------------------------------------------------|--------|
| 61  | Show/hide some fields, depending on other field values                |   ✅ done      |
| 44  | media field type, to easily upload files and store them using spatie/laravel-media |    ✅ done     |
| 41  | Export operation, that would create the PDF/XLS/CSV/JSON using PHP instead of JS; this would allow exports of millions of entries and ability for the developer to easily customize the export process |    🚧  prototype     |
| 35  | list widget / table widget - a component that shows a table of items (like ListOperation), with buttons to edit those entries, that you can include anywhere (eg. on the show page of an entry) |     🚧 draft    |
|  28  | Drag & drop page builder, to allow creating presentation pages or polls out of blocks. |     ❌ prototype, but decided not to do   |
| 23  | an upload field that would work inside repeatable                      |    ✅  done     |
| 20  | Livewire components - allow the admin to create/update/list entries without pageloads |  🚧  draft     |
| 17  | google_maps field type that allows the admin to pick a point on map   |    ✅ done      |
| 16  | dropzone field type, that will allow admins to upload files using AJAX |    ✅ done      |
| 13  | SoftOrHardDelete operation, that would allow an admin to trash / untrash / hard-delete an entry |    ✅ done      |
| 11  | Language Picker - allow the admin to change the language of his admin panel |    ✅ done     |

Almost two years later, we _have_ worked on many of the requested features... but **we haven't finished and launched _all_ of them**. It's time to change that. **We have 3 months... until the end of Nov 2023... to finish more of the above** 💪 So that afterwards, we can do another poll, and ask our users "_what do they want_". And hopefully going forward, we'll do this poll _yearly_, and get more and more feedback from our users.

So here's the plan, from easy to crazy-difficult:
- @promatik will finish the Language Picker - size: M, one week
- @pxpm will finish the ExportOperation - size: L, one month
- @tabacitu will finish a _basic_ Livewire widget for a table (that you can include anywhere) - size: L, one month
- @tabacitu will do the architecture for the main Livewire components (table, form, chart), so that we can start building them out 
    - the archicture itself should be size: L (one month); DDL: Nov 2023;
    - actually finishing all components will be a team effort, size: XXL (3-4 months); DDL: Feb 2024;

The features above are MUST-DOs. They help us reach our goals, by _doing what our users have asked for_. **Let's not get distracted by other cool features!** Some other small features... fine, we can cram them in... but only if _these features are on track_! @Laravel-Backpack/team-members , please keep me accountable. Tell me if we get distracted. If I prioritise some other feature ahead of these, tell me - "_Tabacitu, don't be stupid. Eyes on the prize!_" 😅

Ready to make our users happy?! Let's do it 💪

PS. Of course, _maintenance_ and _marketing_ have nothing to do with the above. Those aren't _features_.

**Comment by ale1981** [comment]
@tabacitu  are these still being worked on? Particularly interested in the list / table widget. Would this mean that a relation table could be added anywhere on the page that would look like a list of that particular model? Currently I have created a custom widget for this but would be interested in how this will work with the core widget.

**Reply by tabacitu** [reply]
Yes we're still working on it @ale1981 ! Regarding the table widget, until we have a Livewire one, you can also use this add-on - https://github.com/izica/relations-widgets-for-backpack - I know a lot of people in our community do, and are happy with it.

**Reply by ale1981** [reply]
@tabacitu, thanks for the reply. I was using that add-on however wasn't happy with a lot of the functionality so ended up creating my own.
